usermods/Temperature/usermod_temperature.h

```diff
@@ -52,80 +82,110 @@ class UsermodTemperature : public Usermod {
     static const char _readInterval[];
     static const char _parasite[];
 
-    //Dallas sensor quick (& dirty) reading. Credit to - Author: Peter Scargill, August 17th, 2013
-    float readDallas() {
-      byte i;
-      byte data[2];
-      int16_t result;                         // raw data from sensor
-      if (!oneWire->reset()) return -127.0f;  // send reset command and fail fast
-      oneWire->skip();                        // skip ROM
-      oneWire->write(0xBE);                   // read (temperature) from EEPROM
-      for (i=0; i < 2; i++) data[i] = oneWire->read();  // first 2 bytes contain temperature
-      for (i=2; i < 8; i++) oneWire->read();  // read unused bytes  
-      result = (data[1]<<4) | (data[0]>>4);   // we only need whole part, we will add fraction when returning
-      if (data[1]&0x80) result |= 0xFF00;     // fix negative value
-      oneWire->reset();
-      oneWire->skip();                        // skip ROM
-      oneWire->write(0x44,parasite);          // request new temperature reading (without parasite power)
-      return (float)result + ((data[0]&0x0008) ? 0.5f : 0.0f);
-    }
+    void Ds18x20Init(void) {
+      ds = new OneWire(temperaturePin);
 
-    void requestTemperatures() {
-      readDallas();
-      lastTemperaturesRequest = millis();
-      waitingForConversion = true;
-      DEBUG_PRINTLN(F("Requested temperature."));
+      Ds18x20Search();
     }
-
-    void readTemperature() {
-      temperature = readDallas();
-      lastMeasurement = millis();
-      waitingForConversion = false;
-      //DEBUG_PRINTF("Read temperature %2.1f.\n", temperature); // does not work properly on 8266
-      DEBUG_PRINT(F("Read temperature "));
-      DEBUG_PRINTLN(temperature);
+    
+    void Ds18x20Search(void) {
```

Comment:
ESP32 may have timing issues with OneWire. Check this routine thoroughly on ESP32 with short and long (3-5m) wires on sensors.